crates/hdx_parser/src/comparison.rs

```diff
@@ -13,25 +13,29 @@ pub enum Comparison {
 
 impl<'a> Parse<'a> for Comparison {
 	fn parse(parser: &mut Parser<'a>) -> Result<Comparison> {
-		Ok(match parser.next() {
-			Token::Delim('=') => Comparison::Equal,
-			Token::Delim('>') => {
-				if let Token::Delim('=') = parser.peek_with(Include::Whitespace) {
+		Ok(match (parser.next().kind(), parser.next().char()) {
```

Comment:
IIRC `next()` advances the position so this wouldn't work, you might need something like this instead:

```suggestion
    let token = parser.next();
		Ok(match (token.kind(), token.char()) {
```